lib/src/utils.dart

```diff
@@ -86,3 +90,219 @@ class MacOSBrightnessOverrideHandler {
     _lastBrightness = currentBrightness;
   }
 }
+
+/// A class that listens for changes to the application's window being the main
+/// window, and notifies listeners.
+class WindowMainStateListener {
+  /// A shared instance of [WindowMainStateListener].
+  static final instance = WindowMainStateListener();
+
+  /// A [NSWindowDelegateHandle], to be used when deiniting the listener.
+  NSWindowDelegateHandle? handle;
+
+  /// Whether the window is currently the main window.
+  bool _isMainWindow = true;
+
+  /// Whether the window is currently the main window.
+  bool get isMainWindow => _isMainWindow;
+
+  /// Notifies listeners when the window’s main state changes.
+  final _windowMainStateStreamController = StreamController<bool>.broadcast();
+
+  /// A stream of the window’s main state. Emits a new value whenever the state
+  /// changes.
+  Stream<bool> get onChangedStream => _windowMainStateStreamController.stream;
+
+  /// Initializes the listener. This should only be called once.
+  void _init() {
+    if (kIsWeb) return;
+    if (!Platform.isMacOS) return;
+
+    _initDelegate();
+    _initIsWindowMain();
+  }
+
+  /// Initializes the [NSWindowDelegate] to listen for main window changes.
+  void _initDelegate() {
+    final delegate = _WindowMainStateListenerDelegate(
+      onWindowDidBecomeMain: () {
+        _isMainWindow = true;
+        _windowMainStateStreamController.add(true);
+      },
+      onWindowDidResignMain: () {
+        _isMainWindow = false;
+        _windowMainStateStreamController.add(false);
+      },
+    );
+    handle = WindowManipulator.addNSWindowDelegate(delegate);
+  }
+
+  /// Initializes the [_isMainWindow] variable.
+  Future<void> _initIsWindowMain() async {
+    _isMainWindow = await WindowManipulator.isMainWindow();
+    _windowMainStateStreamController.add(_isMainWindow);
+  }
+
+  /// Deinitializes the listener.
+  void deinit() {
+    handle?.removeFromHandler();
+  }
+
+  /// A class that listens for changes to the application's window being the
+  /// main window, and notifies listeners.
+  WindowMainStateListener() {
+    _init();
+  }
+}
+
+/// The [NSWindowDelegate] used by [WindowMainStateListener].
+class _WindowMainStateListenerDelegate extends NSWindowDelegate {
+  _WindowMainStateListenerDelegate({
+    required this.onWindowDidBecomeMain,
+    required this.onWindowDidResignMain,
+  });
+
+  /// Called when the window becomes the main window.
+  final void Function() onWindowDidBecomeMain;
+
+  /// Called when the window resigns as the main window.
+  final void Function() onWindowDidResignMain;
+
+  @override
+  void windowDidBecomeMain() => onWindowDidBecomeMain();
+
+  @override
+  void windowDidResignMain() => onWindowDidResignMain();
+}
+
+/// A class that listens to accent color changes.
```

Comment:
```suggestion
/// A class that listens to changes to the user's selected system accent color.
///
/// Native macOS applications respond to such changes immediately.
```
We should also add a sample code snippet demonstrating usage, unless this is something that only happens under the hood.

Reply:
It’s technically meant to only be used under the hood, although exposing it to macos_ui users probably wouldn’t really hurt, either. In both cases, adding code snippets is a good idea, even if they’re meant to only be read only by contributors, so I’ll do that.